src/views/forms/AccountManagementFormsNew/SlippageEditor.tsx

```diff
@@ -0,0 +1,153 @@
+import { useEffect, useRef, useState, type Dispatch, type SetStateAction } from 'react';
+
+import type { NumberFormatValues } from 'react-number-format';
+import styled from 'styled-components';
+
+import { ButtonShape, ButtonSize } from '@/constants/buttons';
+import { STRING_KEYS } from '@/constants/localization';
+
+import { useStringGetter } from '@/hooks/useStringGetter';
+
+import { FormInput } from '@/components/FormInput';
+import { Icon, IconName } from '@/components/Icon';
+import { InputType } from '@/components/Input';
+import { Output, OutputType } from '@/components/Output';
+import { ToggleGroup } from '@/components/ToggleGroup';
+import { WithConfirmationPopover } from '@/components/WithConfirmationPopover';
+
+enum EditorState {
+  Viewing = 'Viewing',
+  Selecting = 'Selecting',
+  Editing = 'Editing',
+}
+
+type ElementProps = {
+  slippage: number;
+  setIsEditing?: Dispatch<SetStateAction<boolean>>;
+  setSlippage: (slippage: number) => void;
+  disabled?: boolean;
+};
+
+export type SlippageEditorProps = ElementProps;
+
+export const SlippageEditor = ({
+  disabled,
+  slippage,
+  setIsEditing,
+  setSlippage,
+}: SlippageEditorProps) => {
+  const percentSlippage = slippage * 100;
+  const [slippageInputValue, setSlippageInputValue] = useState(percentSlippage.toString());
+  const [editorState, setEditorState] = useState(EditorState.Viewing);
+
+  const stringGetter = useStringGetter();
+
+  const toggleGroupRef = useRef<HTMLDivElement>(null);
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    setIsEditing?.(editorState !== EditorState.Viewing);
+
+    if (editorState === EditorState.Selecting) {
+      toggleGroupRef?.current?.focus();
+    } else if (editorState === EditorState.Editing) {
+      inputRef?.current?.focus();
+    }
+  }, [editorState, setIsEditing]);
+
+  const onOpenChange = (isOpen: boolean) => {
+    if (!isOpen) {
+      setEditorState(EditorState.Viewing);
+    }
+  };
+
+  const onCancel = () => {
+    setEditorState(EditorState.Viewing);
+    setSlippageInputValue(percentSlippage.toString());
+  };
+
+  const onChangeSlippage = ({ value }: NumberFormatValues) => {
+    setSlippageInputValue(value);
+  };
+
+  const onConfirmSlippage = () => {
+    setEditorState(EditorState.Viewing);
+    setSlippage(Number(slippageInputValue) / 100);
```

Comment:
why the `/ 100` here? It looks like you multiply by 100 to set the initial `slippageInputValue`

Reply:
Ah nevermind. I see that this is `setSlippage` and not `setSlippageInputValue`.

Reply:
i believe slippage is consumed as a decimal but we want to represent it as a percent in the UI so we multiply by 100 when receiving the state as a prop and divide by 100 when setting the state from the parent component

Reply:
wtf, i swear your followup comment didn't appear on my screen till after i responded. my bad 😮‍💨